akka-actor/src/main/scala/akka/actor/Stash.scala

```diff
@@ -75,6 +75,7 @@ trait UnrestrictedStash extends Actor with StashSupport {
    *  Overridden callback. Prepends all messages in the stash to the mailbox,
    *  clears the stash, stops all children and invokes the postStop() callback.
    */
+  @throws(classOf[Exception])
```

Comment:
Curious, is this required now?

Reply:
Seems so. It's a bit weird, but not unreasonable: strictly speaking AFAICS not having this annotation would mean your own `preRestart` was allowed to throw an exception if you extended `Actor`, but would lose that freedom when also mixing in `StashSupport`. I don't think that was the intention.